### ISSUE TITLE
Double quoting in program wrappers

### ISSUE THREAD

**Issue by mickk-on-cpp** [NixOS/nixpkgs#33625]
## Issue description

It appears a number of program wrappers exhibit some double quoting. Using Mercurial 4.3.2 from unstable as an example:

```bash
#! /nix/store/65l6hr8snf4v823f974k97jc65i7bhvf-bash-4.4-p12/bin/bash -e
export TK_LIBRARY='"/nix/store/gwzzcr61sk5arbi9vzh2c3pnz4k2z930-tk-8.6.6/lib/tk8.6"'
export HG='"/nix/store/6ji054br1di9iwm4md73gr8p12lvp60y-mercurial-4.3.2/bin/hg"'
export PATH='"/nix/store/gwzzcr61sk5arbi9vzh2c3pnz4k2z930-tk-8.6.6/bin"'${PATH:+':'}$PATH
exec -a "$0" "/nix/store/6ji054br1di9iwm4md73gr8p12lvp60y-mercurial-4.3.2/bin/.hg-wrapped"  "${extraFlagsArray[@]}" "$@"
```

Notice the `export HG='"…"'` pattern, which uses double quotes literally. As a result `"$HG"` does not point to the right place, which I assume is unintended. Beyond just Mercurial a naive, brute-force way to gauge the situation is to run e.g.:

```bash
grep "^export\s\+\w\+='\"" -rI /nix/store
```

which as you might expect is time consuming. `rg "^exports\s+\w+='\"" /nix/store` is a less time consuming alternative.

There are also wrappers which have `export FOO=""/path/to/thing""` as a pattern which is less disruptive but doesn’t look correct either.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.73, NixOS, 18.03pre124015.f59a0f7f1a6 (Impala)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(mickk): `""`
 - channels(root): `"nixos-18.03pre124015.f59a0f7f1a6"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

**Comment by vcunat** [comment]
Hmm, we added quoting to `makeWrapper` not long ago, and these must be some forgotten call sites... which we should fix now.

**Comment by orivej** [comment]
I have fixed two cases that were in my store, `hplip` in fe05782a4f5d8fc277238e5c615e5dfc3fdd3cf5, and `mercurialFull` in 2e285a3e6cf23d8df398ad5540d9ad24fb80d033. I have found no cases of `export FOO=""/path/to/thing""`. If you have more examples, please post them.

**Comment by mickk-on-cpp** [comment]
On closer look I forgot to account for empty variables when it comes to double double quotes, e.g. `export FOO=""` (with nothing else on the line). Excluding these that only leaves us with old Mercurial derivations.

For `'"` I have wrapper scripts brought by `mercurialFull` and `neovim`.

**Comment by orivej** [comment]
Fixed `neovim` in f4afbf6e77a047db5d30fb33ca778ac957a06b9d
